### DISCUSSION TITLE
How are you using the SDK?

### DISCUSSION THREAD

**Discussion by dekobon** [nginxinc/ngx-rust#70] [General]
Hi folks,

I'm one of the maintainers of this project and I'd love to get some feedback about the SDK. What are you using it for? What are your pain points? How has it helped you. Have you published any open source modules based on it?

**Comment by nivaldogmelo** [comment]
Hi! I've used in [here](https://github.com/nivaldogmelo/ngx-fault-injection)! I was able to use the SDK for almost everything (Had to use the binding's to get `X-Real-IP` header without retrieving all headers and using a iterator for it.

Currently it's buggy, because I'm using `thread::sleep` and it's blocking nginx worker. I could use the `tokio::sleep` but I believe it would have the same problem and that the ideal solution is to use the Nginx events API, so I'm learning a little bit more about it :smile:

If you have any tips I would love to hear it! Thanks for the project!
